pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -274,6 +287,75 @@ func (pc *PackageControllerClient) IsInstalled(ctx context.Context) bool {
 	return hasResource && err == nil
 }
 
+func formatYamlLine(space, key, value string) string {
+	if value == "" {
+		return ""
+	}
+	return space + key + ": " + value + "\n"
+}
+
+func formatImageResource(resource *v1alpha1.ImageResource, name string) (result string) {
+	if resource.CPU != "" || resource.Memory != "" {
+		result = "    " + name + ":\n"
+		result += formatYamlLine("      ", "cpu", resource.CPU)
+		result += formatYamlLine("      ", "memory", resource.Memory)
+	}
+	return result
+}
+
+func formatCronJob(cronJob *v1alpha1.PackageControllerCronJob) (result string) {
+	if cronJob != nil {
+		result += "cronjob:\n"
+		result += formatYamlLine("  ", "digest", cronJob.Digest)
+		result += formatYamlLine("  ", "repository", cronJob.Repository)
+		result += formatYamlLine("  ", "suspend", strconv.FormatBool(cronJob.Disable))
+		result += formatYamlLine("  ", "tag", cronJob.Tag)
+	}
+	return result
+}
+
+func formatResources(resources *v1alpha1.PackageControllerResources) (result string) {
+	if resources.Limits.CPU != "" || resources.Limits.Memory != "" ||
+		resources.Requests.CPU != "" || resources.Requests.Memory != "" {
+		result += "  resources:\n"
+		result += formatImageResource(&resources.Limits, "limits")
+		result += formatImageResource(&resources.Requests, "requests")
+	}
+	return result
+}
+
+// GetPackageControllerConfiguration returns the default kubernetes version for a Cluster.
+func (pc *PackageControllerClient) GetPackageControllerConfiguration() (result string, err error) {
+	clusterSpec := pc.clusterSpec
+	if clusterSpec == nil || clusterSpec.Packages == nil {
+		return "", nil
+	}
+
+	if clusterSpec.Packages.Controller != nil {
+		result += "controller:\n"
```

Comment:
I am still not comfortable with custom parsing instead of using the classic json marshaller.